deployer/client/bootstrap.go

```diff
@@ -2,34 +2,57 @@ package client
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/coinbase/step/aws"
+	"github.com/coinbase/step/deployer"
 	"github.com/coinbase/step/utils/to"
 )
 
 // Bootstrap takes release information and uploads directly to Step Function and Lambda
-func Bootstrap(states *string, lambda *string, step *string, bucket *string, zip_file_path *string) error {
-	awsc := aws.CreateAwsClients()
+func Bootstrap(release *deployer.Release, zip_file_path *string) error {
+	awsc := &aws.AwsClientsStr{}
 
 	fmt.Println("Preparing Release Bundle")
-	release, err := PrepareReleaseBundle(awsc, states, lambda, step, bucket, zip_file_path)
+	err := PrepareRelease(release, zip_file_path)
+	if err != nil {
+		return err
+	}
+
+	bts, err := fileBytes(zip_file_path)
 	if err != nil {
 		return err
 	}
 
 	fmt.Println("Deploying Step Function")
 	fmt.Println(to.PrettyJSONStr(release))
-	err = release.DeployStepFunction(awsc.SFNClient())
+
+	err = release.DeployStepFunction(awsc.SFNClient(nil, nil, nil))
 	if err != nil {
 		return err
 	}
 
 	fmt.Println("Deploying Lambda Function")
-	err = release.DeployLambda(awsc.LambdaClient())
+
+	err = release.DeployLambdaCode(awsc.LambdaClient(nil, nil, nil), bts)
 	if err != nil {
```

Comment:
in the case that step function deploys and lambda doesn't, is it worth tearing down the step function so there isn't some partial deploy happening?